internal/bootstrap/handlers/version.go

```diff
@@ -85,10 +85,21 @@ func (vv *VersionValidator) BootstrapHandler(
 	}
 
 	url := config.Clients[common.CoreDataClientName].Url() + clients.ApiVersionRoute
-	data, err := clients.GetRequestWithURL(context.Background(), url)
-	if err != nil {
-		logger.Error("Unable to get version of Core Services", "error", err)
-		return false
+	var data []byte
+	var err error
+	for startupTimer.HasNotElapsed() {
+		if data, err = clients.GetRequestWithURL(context.Background(), url); err != nil {
+			logger.Warn(err.Error())
+			select {
+			case <-ctx.Done():
```

Comment:
Humm, not sure we need the select. Not sure why the bootstrap is using that for registry. this is typically needed if you spawn a go func to do the work.  I think it can be simplified to what you did for NewStoreClient.